src/python/zquantum/core/interfaces/ansatz_test.py

```diff
@@ -0,0 +1,73 @@
+class AnsatzTests(object):
```

Comment:
Adressed in [947777](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/947777d7ee2aed4d0e3eef8556e4584511a1e2aa)

Reply:
I must say I disagree. Checking of the return type defies the whole duck-typing thing. Besides, this is why we use typing, right? Analysis of the return type is strictly a responsibility of static type checker, unless some fancy Union stuff is involved.

Reply:
I do agree with that and honestly was conflicted about including it, but it's the main functionality that the `Ansatz` interface promises, so I think it's good to test. Maybe as opposed to strict type check, we can instead make sure the circuit returns has `len(circuit.gates) > 0` and `len(circuit.qubits) > 0` or something along those lines? Maybe also include 

```
if not ansatz.supports_parametrized_circuits:
    assert ansatz.get_symbols() is None # (or assert raises an Error)
```

Reply:
I'll leave it as it is for now, but lets have a discussion about how do we write test.

Reply:
Actually, I've changed it and added some new tests in [ba699a](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/ba699aa6b37dbc9aaa61a47b5a8e73390ca2a7a1) and [2970b7](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/2970b72e2c06afb71e47072cc50746ca29fc17f1).

But I wouldn't go too far with these, since I think specific checks whether the circuit is correct should be done on the side of ansatz implementation.